contracts/multisig-prover/src/contract/execute.rs

```diff
@@ -225,67 +229,90 @@ pub fn update_verifier_set(
     let multisig: multisig::Client =
         client::ContractClient::new(deps.querier, &config.multisig).into();
 
-    let cur_verifier_set = CURRENT_VERIFIER_SET
-        .may_load(deps.storage)
+    let new_verifier_set = make_verifier_set(&deps, &env, &config)?;
+    CURRENT_VERIFIER_SET
+        .save(deps.storage, &new_verifier_set)
         .map_err(ContractError::from)?;
 
-    match cur_verifier_set {
-        None => {
-            // if no verifier set, just store it and return
-            let new_verifier_set = make_verifier_set(&deps, &env, &config)?;
-            CURRENT_VERIFIER_SET
-                .save(deps.storage, &new_verifier_set)
-                .map_err(ContractError::from)?;
-
-            Ok(Response::new()
-                .add_message(multisig.register_verifier_set(new_verifier_set.clone()))
-                .add_message(
-                    coordinator.set_active_verifiers(
-                        new_verifier_set
-                            .signers
-                            .values()
-                            .map(|signer| signer.address.to_string())
-                            .collect::<HashSet<String>>(),
-                    ),
-                ))
-        }
-        Some(cur_verifier_set) => {
-            let new_verifier_set = next_verifier_set(&deps, &env, &config)?
-                .ok_or(ContractError::VerifierSetUnchanged)?;
+    Ok(Response::new()
+        .add_message(multisig.register_verifier_set(new_verifier_set.clone()))
+        .add_message(
+            coordinator.set_active_verifiers(
+                new_verifier_set
+                    .signers
+                    .values()
+                    .map(|signer| signer.address.to_string())
+                    .collect::<HashSet<String>>(),
+            ),
+        ))
 
-            save_next_verifier_set(deps.storage, &new_verifier_set)?;
+    // let cur_verifier_set = CURRENT_VERIFIER_SET
```

Comment:
do we need old code? it's in git anyway

Reply:
Good question: I didn't remove it to be obvious that signer rotation is not implemented. Also the commented out code is the reason why I set some tests as `ignore`

Reply:
@anstylian Yeah, I suspected it too, but better to have a simple short explanation why it's commented out

Reply:
I will remove it